precompiles/xcm/src/lib.rs

```diff
@@ -264,6 +225,21 @@ where
                 fees: fee_multilocation.clone().into(),
                 weight_limit: WeightLimit::Unlimited,
             },
+            SetAppendix(Xcm(vec![DepositAsset {
+                assets: All.into(),
+                beneficiary: MultiLocation {
+                    parents: 1,
+                    interior: X2(
+                        // last() returns the last Juction Enum in Junctions
+                        // for Univeral Location it is the Parachain() variant
+                        *context.last().unwrap(),
+                        AccountId32 {
+                            network: None,
+                            id: origin.into(),
+                        },
+                    ),
+                },
+            }])),
```

Comment:
Why is this needed, in the context of this PR?
Isn't this functionality already covered?

Reply:
This is not really related to the redesign task, but we wanted to provide the functionality so that extra token get deposited into the sender's account on other parachain.

We have allowed users to send custom xcm but there is no harm in including this in `remote_transact()`

Reply:
There is harm actually - you're bringing unnecessary changes to the existing functionality.
This is used by smart contracts, which should (mostly) be immutable functionality-wise.

But this comment can be put aside - highest prio is [this](https://github.com/AstarNetwork/astar-frame/pull/161#discussion_r1236570211).

Reply:
Should I remove this `SetAppendix` or not?

Reply:
Yes, you should remove it.

You gain nothing by adding it, except changing existing SC behavior which might not be desirable at all.